libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/HttpRequestUtils.java

```diff
@@ -43,29 +42,37 @@
             throw new IllegalArgumentException("endpoint");
         }
 
-        return CompletableFuture.supplyAsync(() -> {
-            RequestBody requestBody = RequestBody.create(MediaType.parse("application/json"), payloadBody);
-            OkHttpClient client = new OkHttpClient();
-            String endpointKey = String.format("%s", endpoint.getEndpointKey());
+        ObjectMapper mapper = new ObjectMapper();
+        String endpointKey = String.format("%s", endpoint.getEndpointKey());
+        Response response;
+        JsonNode qnaResponse = null;
+        try {
+            Request request = buildRequest(requestUrl, endpointKey, buildRequestBody(payloadBody));
+            response = this.httpClient.newCall(request).execute();
+            qnaResponse = mapper.readTree(response.body().string());
+            if (!response.isSuccessful()) {
+                String message = new StringBuilder("The call to the translation service returned HTTP status code ")
+                    .append(response.code()).append(".").toString();
+                throw new Exception(message);
+            }
+        } catch (Exception e) {
+            LoggerFactory.getLogger(HttpRequestUtils.class).error("findPackages", e);
+            throw new CompletionException(e);
+        }
 
-            Request request = new Request.Builder().url(requestUrl).header("Authorization", String.format("EndpointKey %s", endpointKey))
-                    .header("Ocp-Apim-Subscription-Key", endpointKey).header("User-Agent", UserAgent.value())
-                    .post(requestBody).build();
+        return CompletableFuture.completedFuture(qnaResponse);
+    }
 
-            Response response;
-            try {
-                response = client.newCall(request).execute();
-                if (!response.isSuccessful()) {
-                    String message = new StringBuilder("The call to the translation service returned HTTP status code ")
-                            .append(response.code()).append(".").toString();
-                    throw new Exception(message);
-                }
-            } catch (Exception e) {
-                LoggerFactory.getLogger(HttpRequestUtils.class).error("findPackages", e);
-                throw new CompletionException(e);
-            }
+    private Request buildRequest(String requestUrl, String endpointKey, RequestBody body) {
+        HttpUrl.Builder httpBuilder = HttpUrl.parse(requestUrl).newBuilder();
+        Request.Builder requestBuilder = new Request.Builder()
```

Comment:
I think here we need to add the body into the request

Reply:
Done